### PR TITLE
Unit Localization Cluster tweaks

### PR DESCRIPTION
**Pull request by lzgrablic02**
#### Problem
* Issues with Unit Localization Cluster

#### Change overview
Renamed XML file to match name in zcl.json and all.xml.
Added explicit ClusterRevision.
Corrected type for TemperatureUnit.
Added FeatureMap.
ZAP regen.

#### Testing
* None

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml [hunk 1]
```diff
@@ -30,10 +30,15 @@ limitations under the License.
...
     <description>Nodes should be expected to be deployed to any and all regions of the world. These global regions
-      may have differing preferences for the units in which values are conveying in communication to a
+      may have differing preferences for the units in which values are conveyed in communication to a
       user. As such, Nodes that visually or audibly convey measurable values to the user need a
       mechanism by which they can be configured to use a user’s preferred unit.</description>
-    <!-- Base data types -->      
-    <attribute side="server" code="0x00" define="TEMPERATURE_UNIT" type="ENUM8" writable="true" optional="true">TemperatureUnit</attribute>
+    <globalAttribute side="either" code="0xFFFD" value="1"/>
+    <!-- Base data types -->
+    <attribute side="server" code="0x00" define="TEMPERATURE_UNIT" type="TempUnit" writable="true" optional="true">TemperatureUnit</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml:38]: TempUnit is not defined anywhere, right?

Comment (lzgrablic02) [reply src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml:38]: It's defined above, in this file.

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml:38]: Oh, indeed. I missed that somehow.  My apologies.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by lzgrablic02** [comment]
Rebased

**Review by andy31415 — APPROVED** [review]

**Comment by lzgrablic02** [comment]
https://github.com/project-chip/connectedhomeip/pull/14258 includes these changes and checks are passing there

**Pull request closed without merging**